variants/adafruit_feather/pins_arduino.h

```diff
@@ -41,4 +41,9 @@
 #define SPI_HOWMANY    (1u)
 #define WIRE_HOWMANY   (1u)
 
+// D pins
+#define D4             (6u)
+#define D5             (7u)
+#define D6             (8u)
+
```

Comment:
Looking at the schematic here https://cdn-learn.adafruit.com/assets/assets/000/116/896/original/adafruit_products_FeatherRP2040_sch.png?1670021606 it looks like there are D7-D13 and D24-D25.  Can you add those here, too?

Reply:
D7-8 are not included, and D9-13, 24,25 are mapped correctly

Reply:
Actually, that points to another problem.  This `#define` doesn't compile at all.  The issue is the variant pins.h for both chips include "../generic/common.h`.  In there they define
````
static const uint8_t D0 = (0u);
static const uint8_t D1 = (1u);
static const uint8_t D2 = (2u);
static const uint8_t D3 = (3u);
static const uint8_t D4 = (4u);
....
static const uint8_t D27 = (27u);
static const uint8_t D28 = (28u);
static const uint8_t D29 = (29u);
````

So your `#define D4 (6u)` ends up making a line that compiles as
````
static const uint8_t (6u) = (4u);
````

I just tried it locally and it does fail as expected.

````
In file included from /home/earle/Arduino/hardware/pico/rp2040/cores/rp2040/Arduino.h:30,
                 from /tmp/arduino_build_712534/sketch/sketch_jul12a.ino.cpp:1:
/home/earle/Arduino/hardware/pico/rp2040/variants/adafruit_feather/pins_arduino.h:45:25: error: expected unqualified-id before numeric constant
   45 | #define D4             (6u)
      |                         ^~
/home/earle/Arduino/hardware/pico/rp2040/variants/adafruit_feather/../generic/common.h:20:22: note: in expansion of macro 'D4'
   20 | static const uint8_t D4 = (4u);
      |                      ^~
/home/earle/Arduino/hardware/pico/rp2040/variants/adafruit_feather/pins_arduino.h:45:25: error: expected ')' before numeric constant
   45 | #define D4             (6u)
      |                        ~^~
/home/earle/Arduino/hardware/pico/rp2040/variants/adafruit_feather/../generic/common.h:20:22: note: in expansion of macro 'D4'
   20 | static const uint8_t D4 = (4u);
      |                      ^~
````

Reply:
What needs to be done is either define all D/A pins and SDA/SCL/etc. (everything in common.h) in each variant (the hacky way) or to rewrite the common to use a predefined constant if defined for each pin.  I.e.
````
...
#ifdef __D4
static const uint8_t D4 = __D4;
#else
static const uint8_t D4 = (4u);
#endif
...
````

There may be a cleaner way of doing this, but off the top of my head I think it would work.

Reply:
Actually, the suggestion was already implemented for the ADC pins!  Been so long I forgot it, but we have a `__PIN_XX` define

````
...
#ifdef __PIN_A3
static const uint8_t A3 = __PIN_A3;
#else
static const uint8_t A3 = (29u);
#endif
...
````

Doing the digital pins in the same way would be relatively small technical debt I think...

Reply:
thanks! I added changes to common.h